docs/vertx-runtime/master.adoc

```diff
@@ -290,6 +258,55 @@ include::topics/secured-mission-resources.adoc[leveloffset=+3]
 //Add-ons
 //Building your app
 
+== Debugging
+
+This sections contains information about debugging your {VertX}&#x2013;based application both in local and remote deployments.
+
+=== Remote Debugging
+:context:
+
+To remotely debug an application, you must first configure it to start in a debugging mode, and then attach a debugger to it.
+
+
+:parameter-debug-property: debug.port
+:parameter-maven-goal: vertx:debug
+include::topics/proc_starting-your-application-locally-in-debugging-mode.adoc[leveloffset=+3]
+:parameter-debug-property!:
+:parameter-maven-goal!:
+
+//:parameter-uberjar-documented:
+//include::topics/proc_starting-an-uberjar-in-debugging-mode.adoc[leveloffset=+3]
+//:parameter-uberjar-documented!:
+
+include::topics/proc_starting-your-application-on-openshift-in-debugging-mode.adoc[leveloffset=+3]
+
+include::topics/proc_attaching-a-remote-debugger-to-the-application.adoc[leveloffset=+3]
+
+=== Debug Logging
+
+{VertX} provides a built-in logging API.
+The default logging implementation for {VertX} uses the `java.util.logging` library that is provided with the {ProductShortName} release of {VertX}.
+Alternatively, {VertX} allows you to implement a different logging framework, for example, link:https://logging.apache.org/log4j/2.x/javadoc.html[_Log4J_^] ({VertX} supports Log4J v1 and v2) or link:https://www.slf4j.org/docs.html[_SLF4J_s^].
```

Comment:
Probably want to say "use a logging framework" instead of "implement a logging framework".